snapcraft/plugins/go.py

```diff
@@ -144,6 +144,15 @@ def _get_local_go_package(self):
             go_package = os.path.basename(os.path.abspath(self.options.source))
         return go_package
 
+    def _get_local_main_packages(self):
```

Comment:
could you please move this definition after it was called the first time?

after `def build`